test/typecheck/fail/config_non_bit.json

```diff
@@ -0,0 +1,3 @@
+{
+    "k" : [true, false, false, true, "not a bit"]
```

Comment:
That doesn't look like a fun bit vector syntax! Could we also support Sail bit vector literals? I.e. `"0x1234"` and `"0b0101"`